packages/l10n/src/utils.ts

```diff
@@ -20,15 +27,18 @@ export function getSupportedLocale(locale) {
 
 // given a currenyCode and a list of currencies with the following shape
 // [{ label, symbol }], we return the symbol.
-export const getSymbolFromCurrency = (currencyCode, currencies) => {
+export const getSymbolFromCurrency = (
+  currencyCode: string,
+  currencies: Currencies
+) => {
   if (currencies[currencyCode] && currencies[currencyCode].symbol) {
     return currencies[currencyCode].symbol;
   }
   return '';
 };
 
 // From https://github.com/acdlite/recompose/blob/master/src/packages/recompose/getDisplayName.js
-export const getDisplayName = Component => {
+export const getDisplayName = (Component: React.ComponentType<any>) => {
```

Comment:
Is the `any` necessary or can we leave it out?

Reply:
```
packages/l10n/src/create-l10n-injector/create-l10n-injector.tsx:24:11 - error TS2345: Argument of type 'ComponentType<Props>' is not assignable to parameter of type 'ComponentType<{}>'.
  Type 'ComponentClass<Props, any>' is not assignable to type 'ComponentType<{}>'.
    Type 'ComponentClass<Props, any>' is not assignable to type 'ComponentClass<{}, any>'.
      Types of property 'getDerivedStateFromProps' are incompatible.
        Type 'GetDerivedStateFromProps<Props, any> | undefined' is not assignable to type 'GetDerivedStateFromProps<{}, any> | undefined'.
          Type 'GetDerivedStateFromProps<Props, any>' is not assignable to type 'GetDerivedStateFromProps<{}, any>'.
            Types of parameters 'nextProps' and 'nextProps' are incompatible.
              Type 'Readonly<{}>' is not assignable to type 'Readonly<Props>'.

24           WrappedComponent
             ~~~~~~~~~~~~~~~~```

Reply:
I think it's necessary 🤔

Reply:
I don't think `any` is correct here. You actually don't want to allow to accept any from my view. I'd say `Props extends object` in these cases. At least that's what I ended up doing at the flopper.

Reply:
Hmm "object" is also not "safe".

I think we need to parametrize the `getDisplayName` function as well, and pass the generic type to it when we call it

```ts
export function getDisplayName<T>(Component: React.ComponentType<T>) {}
```

Reply:
Not sure. Now I could type `T` to a number. That's what felt odd to me back then and why I didn't allow having the consumer pass the generic.

Reply:
Sure, although it's an internal function.

What you're proposing is instead something like this?

```ts
function getDisplayName<Props extends object>(Component: React.ComponentType<Props>) {}
```